File: tests/rosetta/transpiler/FS/checkpoint-synchronization-1.fs
// Generated 2025-07-27 23:36 +0700

let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let mutable partList: string array = [|"A"; "B"; "C"; "D"|]
let mutable nAssemblies: int = 3
for cycle in 1 .. ((nAssemblies + 1) - 1) do
    printfn "%s" ("begin assembly cycle " + (string cycle))
    for p in partList do
        printfn "%s" (p + " worker begins part")
    for p in partList do
        printfn "%s" (p + " worker completes part")
    printfn "%s" (("assemble.  cycle " + (string cycle)) + " complete")
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)

File: tests/rosetta/transpiler/Java/checkpoint-synchronization-1.java
public class Main {
    static String[] partList;
    static int nAssemblies;

    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            partList = ((String[])(new String[]{"A", "B", "C", "D"}));
            nAssemblies = 3;
            for (int cycle = 1; cycle < (nAssemblies + 1); cycle++) {
                System.out.println("begin assembly cycle " + _p(cycle));
                for (String p : partList) {
                    System.out.println(p + " worker begins part");
                }
                for (String p : partList) {
                    System.out.println(p + " worker completes part");
                }
                System.out.println("assemble.  cycle " + _p(cycle) + " complete");
            }
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }

    static String _p(Object v) {
        return v != null ? String.valueOf(v) : "<nil>";
    }
}
